frontend/src/metabase-lib/moving-average.ts

```diff
@@ -0,0 +1,139 @@
+import { t } from "ttag";
+
+import { breakoutColumn, breakouts } from "./breakout";
+import { isTemporal } from "./column_types";
+import { expressionClause, withExpressionName } from "./expression";
+import { describeTemporalUnit, displayInfo } from "./metadata";
+import { temporalBucket } from "./temporal_bucket";
+import type { AggregationClause, ExpressionClause, Query } from "./types";
+
+function movingAverage(
+  clause: AggregationClause | ExpressionClause,
+  offset: number,
+  start: number,
+) {
+  const clauses = [];
+  for (let i = 0; i > offset; i--) {
+    const period = start + i;
+
+    if (period === 0) {
+      clauses.push(clause);
+    } else {
+      clauses.push(expressionClause("offset", [clause, period]));
+    }
+  }
+
+  const newClause = expressionClause("/", [
+    expressionClause("+", clauses),
+    Math.abs(offset),
+  ]);
+
+  return newClause;
+}
+
+export function movingAverageClause(
```

Comment:
maybe use an object instead, like we do for `...filterParts` etc? `MovingAverageClauseParts`

Reply:
I did it like this to be consistent [with Kamil's work on offsets](https://github.com/metabase/metabase/blob/master/frontend/src/metabase-lib/offset.ts#L10). What's the benefit of using `...Parts`, if we want to turn it into a clause afterwards?

Adding a display name to expression parts is not possible it seems and building and expression clause from `ExpressionParts` using `Lib.expressionClause` is also showing TypeScript errors for me.

Reply:
Parts was just a convention for an interface name. So instead of passing 5 args you would pass an object. There is nothing else being suggested here.

Reply:
Ah, yeah that makes sense!